test/parallel_api/ranges/std_ranges_zip_view.pass.cpp

```diff
@@ -99,6 +99,45 @@ main()
         EXPECT_TRUE(std::get<0>(zip_view_sort[i]) == max_n - 1 - i && std::get<1>(zip_view_sort[i]) == max_n - 1 - i,
             "Wrong effect for oneapi::dpl::ranges::sort with zip_view.");
     }
+#if TEST_DPCPP_BACKEND_PRESENT
+    {
+    const char* err_msg = "Wrong effect for oneapi::dpl::ranges::sort with zip_view and a device policy."
+
+    const int n = big_size;
+    std::vector<int> vals(n), keys(n);
+
+    //test with random number and projection usage
+    std::default_random_engine gen{std::random_device{}()};
+    std::uniform_real_distribution<float> dist(0.0, 100.0);
+
+    std::generate(vals.begin(), vals.end(), [&] { return dist(gen); });
+    std::generate(keys.begin(), keys.end(), [&] { return dist(gen); });
+
+    std::vector<int> vals_exp(vals);
+    std::vector<int> keys_exp(keys);
+
+    auto exec = dpcpp_policy();
+    using Policy = decltype(exec);
+    {
+        usm_subrange<T> cont_vals(exec, vals.data(), n);
+        usm_subrange<T> cont_keys(exec, vals.data(), n);
+        auto view_vals = cont_vals();
+        auto view_keys = cont_keys();
+        auto view_s = dpl_ranges::views::zip(view_vals, view_keys)
+
+        //call Range based sort with a device policy
+        dpl_ranges::sort(exec, view_s, std::ranges::greater{}, [](const auto& a) { return std::get<1>(a);});
```

Comment:
Let's use stable sort here and in the reference call for a predictable value order.